src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -745,6 +746,11 @@ def resample(self, timestamp: datetime) -> Sample[Quantity]:
         # So if we need more performance beyond this point, we probably need to
         # resort to some C (or similar) implementation.
         relevant_samples = list(itertools.islice(self._buffer, min_index, max_index))
+        if not relevant_samples:
+            _logger.warning(
+                "No relevant samples found for component: %s",
+                component_name,
+            )
```

Comment:
I think warning makes sense, because somebody is expecting data that's not showing up.  But it would also lead to a lot of noise when a component is under maintenance, etc.  I wonder if we can just log once when an outage starts and once when it ends

Reply:
This is a much more general issue, this applies to a lot of issues. Eventually I think we really need an alerting mechanism, not sure if the notifications service will be used for that. Looking at logs should be a very low level thing, users should just get notified when something went wrong.

Services like sentry (and probably any observability service out there) can set alarms when a particular type of warning (or log message) if firing  for the first time, or if it is firing a number of times per second, or per hour, etc. and you can silence alerts also for some amount of times (when stuff is in maintenance for example).

At some point we'll need to have this kind of flexibility. For now I think warning every time is a considerable improvement. If it is trivial to do the log when it starts and when it ends, we could add it too, but I wouldn't do it if it makes the code much more complicated.